dsl-reference.md

```diff
@@ -30,6 +30,27 @@
     - [Try](#try)
     - [Wait](#wait)
   + [Flow Directive](#flow-directive)
+  + [Lifecycle Events](#lifecycle-events)
+    - [Workflow Lifecycle Events](#workflow-lifecycle-events)
+      + [Workflow Started](#workflow-started-event)
+      + [Workflow Suspended](#workflow-suspended-event)
+      + [Workflow Resumed](#workflow-resumed-event)
+      + [Workflow Correlation Started](#workflow-correlation-started-event)
+      + [Workflow Correlation Completed](#workflow-correlation-completed-event)
+      + [Workflow Cancelled](#workflow-cancelled-event)
+      + [Workflow Faulted](#workflow-faulted-event)
+      + [Workflow Completed](#workflow-completed-event)
+      + [Workflow Status Changed](#workflow-status-changed-event)
+    - [Task Lifecycle Events](#task-lifecycle-events)
+      + [Task Created](#task-created-event)
+      + [Task Started](#task-started-event)
+      + [Task Suspended](#task-suspended-event)
+      + [Task Resumed](#task-resumed-event)
+      + [Task Retried](#task-retried-event)
+      + [Task Cancelled](#task-cancelled-event)
+      + [Task Faulted](#task-faulted-event)
+      + [Task Completed](#task-completed-event)
+      + [Task Status Changed](#task-status-changed-event)
```

Comment:
Status, as I understood it, only applies to the worklow as a whole
https://github.com/serverlessworkflow/specification/blob/main/dsl.md#status-phases
I

Reply:
Not in my opinion. Task status is extremely important in most, if not all workflow execution scenarios. Think of a long running flow which you represent in a UI. Just having flow events will let your users know the flow has started or ended, but whatever is in the middle will be unknown to them, potentially for hours or even days at a time.

Reply:
ok, then we also need to edit that section to make it clear phase status applies both for workflow and task (I implemented it just for workflows because there was not hint on that section)
Note that I also implemented task created and task completed, because I concur on the usefulness for users, which is unclert to me is that they need more task created and task completed ;)

Reply:
> ok, then we also need to edit that section to make it clear phase status applies both for workflow and task

I edited it in the PR, because as you said it only applied to workflows beforehand. Have you checked the updated section in the PR? I think it's clear, but I'd be happy to update it as you see fit!

Reply:
> is unclert to me is that they need more than task created and task completed

Well, I see what you mean, but having at least `canceled` and `faulted` seemed reasonable. Let's discuss it in today's daily!

Reply:
@cdavernas @JBBianchi as dicussed in Slack, I think that if we have specific event for every phase change (both for workflow an task) we do not really need the status change one (or the other way around, if we have the status change, we do not need the specific ones)
Having only the specific ones (created, started, faulted, completed) covers all the status phase changes, so a user just interested in the fact that the status has changed, can consume them and ignore the cloud event data (thats the beauty of the CloudEvent structure, the same event can be used in different ways depending on the consumer needs)

Reply:
The status change event is a generic event which is in my personal experience critically useful. As a matter of fact, users who want to have lifecycle notifications but that are not interested in event-specific data are usually going for those kind of events. While not necessary per say, they are an absolute plus for many users. They are usually fired after any other events, to communicate in a generic fashion with the outside world.

Reply:
I think users interested just in status change can subscribe to all event types and just check the type CE attribute to find out which phase changed. They can completely ignore (and with CE they do not need to deserialize) the data field (thats why I mention the beauty of CE)
That way we avoid publishing two different types of events for the same ocurrence (that the task changed the status from one state to another, with the current approach, we will be sending two events: the specific and the generic one, Im claiming that the specific can be consumed generically)